index.js

```diff
@@ -244,22 +271,43 @@ module.exports = function (schema, options) {
             this.deletedBy = deletedBy;
         }
 
+        if (schema.path('deletedId')) {
+            this.deletedId = deletedId || deletedBy; // Use deletedId if provided, fall back to deletedBy
+        }
+
         if (options.validateBeforeDelete === false) {
             return this.save({ validateBeforeSave: false }, cb);
         }
 
         return this.save(cb);
     };
 
-    schema.statics.delete =  function (conditions, deletedBy, callback) {
-        if (typeof deletedBy === 'function') {
-            callback = deletedBy;
-            conditions = conditions;
-            deletedBy = null;
+    schema.statics.delete =  function (conditions, params, callback) {
+        let deletedBy = null;
+        let deletedId = null;
+
+
+        if (typeof params === 'function') {
+            callback = params;
+            params = {};
         } else if (typeof conditions === 'function') {
             callback = conditions;
             conditions = {};
-            deletedBy = null;
+            params = {};
+        } else if (typeof params === 'string' || typeof params === 'object' || (params && params._bsontype === 'ObjectID')) {
+            deletedBy = params;
+            params = {};
+        } else if (!params) {
+            params = {};
+        }
```

Comment:
🤔 If you do the default in the prototype you can simplify this
```suggestion
        }
```